chess2.cpp

```diff
@@ -0,0 +1,882 @@
+#include "chess2.h"
+#include<iostream>
+
+int Chess2::perform(str Move) {
+    int original;
+    original = board[Move[2] - 48][Move[3] - 48];
+    board[Move[2] - 48][Move[3] - 48] = board[Move[0] - 48][Move[1] - 48];
+    board[Move[0] - 48][Move[1] - 48] = 0;
+    return original;
+}
+
+str Chess2::push(int row, int col, int desrow, int descol) {
+    using std::to_string;
+    str mystr = to_string(row) + to_string(col) + to_string(desrow) + to_string(descol);
+    return mystr;
+}
+
+str Chess2::computer_move(unsigned short int depth) {
+    str bestmove;
+    bestmove = miniMaxroot(depth, false);
+    perform(bestmove);
+    return bestmove;
+}
+
+buff Chess2::getallmoves(bool turn) {
+    int original = 0;
+    pseudomoves.clear();
+    legal_moves.clear();
+    if (turn == true) {
+        for (int i = 0; i < 8; i++) {
+            for (int j = 0; j < 8; j++) {
+                if (!board[i][j]) continue;
+                if (board[i][j] == 1) getpawnmoves(true, i, j);
```

Comment:
SInce this will be moved to a different class , so class reference can be here with position and move